xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/listener/AutomaticRedirectCreatorListener.java

```diff
@@ -71,11 +84,34 @@ public void onEvent(Event event, Object source, Object data)
             }
             if (autoRedirect) {
                 DocumentRenamedEvent documentRenamedEvent = (DocumentRenamedEvent) event;
-                this.logger.info("Creating automatic redirect from [{}] to [{}].",
-                    documentRenamedEvent.getSourceReference(), documentRenamedEvent.getTargetReference());
+                this.logger.info(CREATING_AUTOMATIC_REDIRECT_FROM_TO, documentRenamedEvent.getSourceReference(),
+                    documentRenamedEvent.getTargetReference());
                 this.modelBridge.createRedirect(documentRenamedEvent.getSourceReference(),
                     documentRenamedEvent.getTargetReference());
             }
+        } else if (event instanceof DocumentDeletedEvent && this.jobContext.getCurrentJob() instanceof DeleteJob) {
+            DeleteJob job = (DeleteJob) this.jobContext.getCurrentJob();
+            DeleteRequest request = (DeleteRequest) job.getRequest();
+            DocumentDeletedEvent documentDeletedEvent = (DocumentDeletedEvent) event;
+
+            // For case when the delete affects also the child pages, only the root document should have a redirect
+            // created.
```

Comment:
Why? Note that a delete job request can include a list of documents that have no parent-child relation. You simply want to delete A, B and C, unrelated documents. In that case ``getCommonParent()`` won't return what you expect. So there's no concept of a 'root' document for a delete action.

Reply:
So the `newBacklinkTarget` should be taken into consideration for all the documents of a delete request? Couldn't this be an unexpected outcome when selecting `affectChildren`? (although an warning can be added to the UI)